### PR TITLE
feat: add Lucide react icons

### PR DESCRIPTION
**Pull request by rssilva**
## Description of the change
- Add Lucide icons as dependency and a small subset of icons from it

Screen Shot 2023-05-18 at 13 19 51

## Type of change

- [x] New feature (non-breaking change that adds functionality)

### Development

- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.

### CODE REVIEW

File: packages/components/src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -42,8 +44,16 @@ const Icon = React.forwardRef<HTMLDivElement, IconProps>(
...
+    let RenderedIcon = undefined;
+
+    if (isHeroIcon(icon)) {
+      // eslint-disable-next-line import/namespace
+      RenderedIcon = HeroOutlineIcons[icon];
+    }
+
+    if (isLucideIcon(icon)) {
+      RenderedIcon = LucideIcons[icon];
+    }
```
Comment (RohitRajendran) [inline packages/components/src/components/Icon/Icon.tsx:56]: No need to check the second `if` if the first `if` is satisfied.
```suggestion
    if (isHeroIcon(icon)) {
      // eslint-disable-next-line import/namespace
      RenderedIcon = HeroOutlineIcons[icon];
    } else if (isLucideIcon(icon)) {
      RenderedIcon = LucideIcons[icon];
    }
```

File: packages/components/src/components/Icon/Icon.tsx [hunk 2]
```diff
@@ -60,7 +70,9 @@ const Icon = React.forwardRef<HTMLDivElement, IconProps>(
           aria-label={title}
           as={RenderedIcon}
           display="block"
+          h={size}
           verticalAlign="middle"
+          w={size}
```
Comment (RohitRajendran) [inline packages/components/src/components/Icon/Icon.tsx:75]: Just for my understanding, why was this change necessary?

File: packages/components/src/components/Icon/isHeroIcon.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import * as HeroOutlineIcons from "@heroicons/react/24/outline";
+import { IconName } from "./IconName";
+
+export const isHeroIcon = (
+  name: IconName
+): name is keyof typeof HeroOutlineIcons => {
```
Comment (RohitRajendran) [inline packages/components/src/components/Icon/isHeroIcon.ts:6]: Isn't this just returning a boolean? Could you help me understand `name is keyof typeof HeroOutlineIcons`?

File: packages/components/src/components/Icon/HeroIconName.ts
Comment (RohitRajendran) [inline packages/components/src/components/Icon/HeroIconName.ts:1]: What do you think about moving the new files into subfolders like `/types` and `/utils` or something to make the file structure a little cleaner and easier to parse? It would make it easier to find and pick out Icon.tsx as the main component file when looking at the folder.

File: .changeset/popular-plums-deny.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@localyze-pluto/components": minor
+---
+
+Add a Lucide Icons as dependency with a specific subset available
```
Comment (RohitRajendran) [inline .changeset/popular-plums-deny.md:5]: We could be a little more descriptive in this changeset and list the new icon options available.

File: packages/components/src/components/Icon/IconName.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+import { HeroIconName } from "./HeroIconName";
+import { LucideIconName } from "./LucideIconName";
+
+export type IconName = HeroIconName | LucideIconName;
```
Comment (RohitRajendran) [inline packages/components/src/components/Icon/IconName.ts:4]: There's a few other components that currently use heroicons: Badge, Button, Input. Shouldn't they be updated to have all the same icon options?

File: packages/components/src/components/Icon/Icon.tsx (hunk 2 above)
Comment (rssilva) [reply packages/components/src/components/Icon/Icon.tsx:75]: if we don't apply those, lucide icons will have a default value of 24 fitting the container

Screen Shot 2023-05-19 at 15 29 12

File: packages/components/src/components/Icon/isHeroIcon.ts (hunk 3 above)
Comment (rssilva) [reply packages/components/src/components/Icon/isHeroIcon.ts:6]: I believe this is a safe(r) method to check if an item is present as key of an object? https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

File: packages/components/src/components/Icon/IconName.ts (hunk 5 above)
Comment (rssilva) [reply packages/components/src/components/Icon/IconName.ts:4]: oh! Good call!

File: packages/components/package.json [hunk 6]
```diff
@@ -51,6 +51,7 @@
     "ariakit": "^2.0.0-next.43",
     "html-react-parser": "^3.0.9",
     "lodash": "^4.17.21",
+    "lucide-react": "^0.218.0",
```
Comment (RohitRajendran) [inline packages/components/package.json:54]: Should this be in both `dependencies` and `peerDependencies`? At least based on how `@heroicons/react`, not sure if that's actually needed.

File: .changeset/popular-plums-deny.md [hunk 7]
```diff
@@ -2,4 +2,4 @@
 "@localyze-pluto/components": minor
 ---
 
-Add a Lucide Icons as dependency with a specific subset available
+Add a Lucide Icons as dependency with a specific subset available: baggage-claim, fingerprint, heart-handshake, palette, slack
```
Comment (RohitRajendran) [inline .changeset/popular-plums-deny.md:5]: Won't we now have `fingerprint` from both Lucide and HeroIcons? Shouldn't we consolidate on 1 so that it's not inconsistently used?

Comment (rssilva) [reply .changeset/popular-plums-deny.md:5]: good call!

File: packages/components/package.json (hunk 6 above)
Comment (RohitRajendran) [reply packages/components/package.json:54]: Ah, my bad!
